cluster/swarm/cluster.go

```diff
@@ -216,7 +216,12 @@ func (c *Cluster) RemoveContainer(container *cluster.Container, force, volumes b
 // RemoveNetwork removes a network from the cluster
 func (c *Cluster) RemoveNetwork(network *cluster.Network) error {
 	err := network.Engine.RemoveNetwork(network)
-	c.refreshNetworks()
+	if err == nil {
+
+	}
+	for _, engine := range c.engines {
+		engine.DeleteNetwork(network)
```

Comment:
Maybe it would be better to call the engine.RemoveNetwork function so we have a single entry point for removing networks and leave the DeleteNetwork as private one?